react/react1/week3/my-app/src/TodoList.js

```diff
@@ -0,0 +1,168 @@
+
+      return item;
+    });
+
+    setList(newList);
+  };
+  //editItem function will map all the list items to check a specific list item is prompted for editing.
+  const editItem = (id) => {
+    const newList = list.map((item) => {
+      if (item.id === id) {
+        const updatedItem = {
+          ...item,
+          isEditing: !item.isEditing,
+        };
+        //if editing is true than the corresponding state will set itself to the edited description
+        setInputEditDescription(item.description);
+        return updatedItem;
+      }
+
+      return item;
+    });
+
+    setList(newList);
+  };
+  // updateItem will map the list items to verify if editing is true than it will update the specific item
+  const updateItem = (id) => {
+    const newList = list.map((item) => {
+      if (item.id === id) {
+        const updatedItem = {
+          ...item,
+          description: inputEditDescription,
+          isEditing: !item.isEditing,
+        };
+
+        return updatedItem;
+      }
+
+      return item;
+    });
+
+    setList(newList);
+  };
+  // deleteTodo function will filter the items and delete only the specific item
+  const deleteTodo = (id) => {
+    setList((prev) => {
+      return prev.filter((item) => item.id !== id);
+    });
+  };
+
+  const todoItems = list.map((todo) => (
+    <FancyBorder key={todo.id}>
+      <span
+        className={
+          todo.isEditing ? "notEditing" : todo.isChecked ? "checked-item" : ""
+        }
+      >
+        {todo.description} --
+        {todo.deadline}
+      </span>
+      <input
+        type="text"
+        id="todoitem"
+        placeholder="Todo Item"
+        className={todo.isEditing ? "editing" : "notEditing"}
+        value={inputEditDescription}
+        onChange={(e) => setInputEditDescription(e.target.value)}
+      ></input>
+      {deleteTodo}
+      <input type="checkbox" onChange={(e) => toggleClass(todo.id)}></input>
+      <button onClick={() => deleteTodo(todo.id)}>Delete</button>
+      <button
+        onClick={() => editItem(todo.id)}
+        className={todo.isEditing ? "notEditing" : "editing"}
+      >
+        Edit
+      </button>
+      <button
+        onClick={() => updateItem(todo.id)}
+        className={todo.isEditing ? "editing" : "notEditing"}
+      >
+        Update
+      </button>
+    </FancyBorder>
+  ));
+
+  const FormPart = (
+    <FormBorder>
+      <div>
+        <form>
+          <label>Todo Description</label>
+          <input
+            type="text"
+            id="todoitem"
+            placeholder="Todo Item"
+            value={inputDescription}
+            onChange={(e) => setInputDescription(e.target.value)}
+          ></input>
+          <label>Deadline</label>
+          <input
+            type="date"
+            id="deadline"
+            value={inputDeadline}
+            onChange={(e) => setInputDeadline(e.target.value)}
+          ></input>
+        </form>
+      </div>
+    </FormBorder>
+  );
```

Comment:
I would encourage you to also use individual components here, like I mentioned in the feedback for the **time-registration** application :)

Reply:
same problem here. not sure about parameter props. then which state should I use use inside this component. when I use inputdescription state and inputdeadline. it says it is undefine.